x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/retention/AbstractExpiredJobDataRemover.java

```diff
@@ -84,15 +87,15 @@ private WrappedBatchedJobsIterator newJobIterator() {
         return new WrappedBatchedJobsIterator(jobsIterator);
     }
 
-    abstract void calcCutoffEpochMs(String jobId, long retentionDays, ActionListener<Long> listener);
+    abstract void calcCutoffEpochMs(String jobId, long retentionDays, ActionListener<Tuple<Long, Long>> listener);
```

Comment:
It might be worth replacing the tuple with a struct-like class that better explains what each value is.